src/System.IO.Pipelines/tests/PipeReaderStreamTests.nonnetstandard.cs

```diff
@@ -0,0 +1,306 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Pipelines.Tests
+{
+    public class PipeReaderStreamTests
+    {
+        public delegate Task<int> ReadAsyncDelegate(Stream stream, byte[] data);
+
+        [Theory]
+        [MemberData(nameof(ReadCalls))]
+        public async Task ReadingFromPipeReaderStreamReadsFromUnderlyingPipeReader(ReadAsyncDelegate readAsync)
+        {
+            byte[] helloBytes = Encoding.ASCII.GetBytes("Hello World");
+            var pipe = new Pipe();
+            await pipe.Writer.WriteAsync(helloBytes);
+            pipe.Writer.Complete();
+
+            var stream = new PipeReaderStream(pipe.Reader);
+
+            var buffer = new byte[1024];
+            int read = await readAsync(stream, buffer);
+
+            Assert.Equal(helloBytes, buffer.AsSpan(0, read).ToArray());
+            pipe.Reader.Complete();
+        }
+
+        [Theory]
+        [MemberData(nameof(ReadCalls))]
+        public async Task AsStreamReturnsPipeReaderStream(ReadAsyncDelegate readAsync)
+        {
+            byte[] helloBytes = Encoding.ASCII.GetBytes("Hello World");
+            var pipe = new Pipe();
+            await pipe.Writer.WriteAsync(helloBytes);
+            pipe.Writer.Complete();
+
+            Stream stream = pipe.Reader.AsStream();
+
+            var buffer = new byte[1024];
+            int read = await readAsync(stream, buffer);
+
+            Assert.Equal(helloBytes, buffer.AsSpan(0, read).ToArray());
+            pipe.Reader.Complete();
+        }
+
+        [Fact]
+        public async Task ReadingWithSmallerBufferWorks()
+        {
+            byte[] helloBytes = Encoding.ASCII.GetBytes("Hello World");
+            var pipe = new Pipe();
+            await pipe.Writer.WriteAsync(helloBytes);
+            pipe.Writer.Complete();
+
+            Stream stream = pipe.Reader.AsStream();
+
+            var buffer = new byte[5];
+            int read = await stream.ReadAsync(buffer);
+
+            Assert.Equal(5, read);
+            Assert.Equal(helloBytes.AsSpan(0, 5).ToArray(), buffer.AsSpan(0, read).ToArray());
```

Comment:
Assert.Equal should support IEnumerable arguments.

Reply:
Spans aren't IEnumerable.

Reply:
It shit the bed for Span.

Reply:
Neither Span and Memory are IEnumerable

Reply:
Right, my bad.